source/common/quic/platform/quiche_mem_slice_impl.h

```diff
@@ -25,17 +25,19 @@ class QuicheMemSliceImpl {
 
   ~QuicheMemSliceImpl();
 
-  // Constructs a QuicheMemSliceImpl by taking ownership of the memory in |buffer|.
+  // Constructs a QuicheMemSliceImpl by taking ownership of the memory in `buffer`.
   QuicheMemSliceImpl(quiche::QuicheBuffer buffer);
   QuicheMemSliceImpl(std::unique_ptr<char[]> buffer, size_t length);
 
-  // Constructs a QuicheMemSliceImpl from a Buffer::Instance with first |length| bytes in it.
-  // Data will be moved from |buffer| to this mem slice.
-  // Prerequisite: |buffer| has at least |length| bytes of data and not empty.
+  // Prerequisite: `buffer` must be non-empty, and its first slice must have
+  // length `length`.
+  // Constructs a QuicheMemSliceImpl, copies `length` bytes into it from
```

Comment:
I *think* Envoy::Buffer::Instance::move() should have been moving the slice given that `length` matches the 1st slice length and single_slice_buffer_ is newly constructed. Did I miss anything?

Reply:
Oh sorry, you are right, the slice is moved at:

https://github.com/envoyproxy/envoy/blob/9ee0f887986da2e116685ab43d86e8678774bb3b/source/common/buffer/buffer_impl.cc#L343-L344

and then

https://github.com/envoyproxy/envoy/blob/9ee0f887986da2e116685ab43d86e8678774bb3b/source/common/buffer/buffer_impl.cc#L307.